From 76f1415ed003dae01dd7451107851d67fae3e371 Mon Sep 17 00:00:00 2001
From: Mathieu Malaterre <mathieu.malaterre@gmail.com>
Date: Tue, 12 Apr 2011 06:48:26 +0200
Subject: This commit should fix the isse only seen in x86/armel platform

serna fails to build reporting a message such as:
error: non-placement deallocation function / error: selected for placement delete
---
 serna/pyapi/common/SString.sip |    2 +-
 serna/sapi/common/SString.h    |    2 +-
 sfworks/common/Debug.h         |    2 +-
 sfworks/common/FlexString.h    |    2 +-
 sfworks/common/FreeStorage.h   |    2 +-
 sfworks/common/IterTraits.h    |    2 +-
 sfworks/common/RefCntStorage.h |    2 +-
 sfworks/common/SubAllocator.h  |    2 +-
 sfworks/common/impl/Debug.cxx  |    4 ++--
 9 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/serna/pyapi/common/SString.sip b/serna/pyapi/common/SString.sip
index 98bcb43..354497c 100644
--- a/serna/pyapi/common/SString.sip
+++ b/serna/pyapi/common/SString.sip
@@ -179,7 +179,7 @@ using namespace SernaApi;
 
 public:
     typedef Char                value_type;
-    typedef unsigned int        size_type;
+    typedef std::size_t         size_type;
     typedef value_type&         reference;
     typedef const value_type&   const_reference;
     typedef value_type*         iterator;
diff --git a/serna/sapi/common/SString.h b/serna/sapi/common/SString.h
index b442d68..86085f4 100644
--- a/serna/sapi/common/SString.h
+++ b/serna/sapi/common/SString.h
@@ -47,7 +47,7 @@ namespace SernaApi {
 class SAPI_EXPIMP SString {
 public:
     typedef Char                value_type;
-    typedef unsigned int        size_type;
+    typedef std::size_t         size_type;
     typedef value_type&         reference;
     typedef const value_type&   const_reference;
     typedef value_type*         iterator;
diff --git a/sfworks/common/Debug.h b/sfworks/common/Debug.h
index b4315fa..43826e0 100644
--- a/sfworks/common/Debug.h
+++ b/sfworks/common/Debug.h
@@ -97,7 +97,7 @@ class DebugOutStream;
 
 class COMMON_DEBUG_EXPIMP Debug {
 public:
-    typedef unsigned int size_type;
+    typedef std::size_t size_type;
 
     enum { INDENT = 2 };
 
diff --git a/sfworks/common/FlexString.h b/sfworks/common/FlexString.h
index 72d557c..d0cdc42 100644
--- a/sfworks/common/FlexString.h
+++ b/sfworks/common/FlexString.h
@@ -71,7 +71,7 @@ public:
     typedef typename traits_type::char_type value_type;
     typedef A allocator_type;
     typedef Storage storage_type;
-    typedef unsigned int size_type;
+    typedef size_t size_type;
     typedef typename A::difference_type difference_type;
 
     typedef typename A::reference reference;
diff --git a/sfworks/common/FreeStorage.h b/sfworks/common/FreeStorage.h
index ce26184..f37d73a 100644
--- a/sfworks/common/FreeStorage.h
+++ b/sfworks/common/FreeStorage.h
@@ -54,7 +54,7 @@
 
 namespace StringPrivate {
 
-typedef unsigned int size_type;
+typedef std::size_t size_type;
 
 template<typename E> struct Data {
     size_type   capacity_;  // length of allocated memory block
diff --git a/sfworks/common/IterTraits.h b/sfworks/common/IterTraits.h
index 41c109a..a9d100e 100644
--- a/sfworks/common/IterTraits.h
+++ b/sfworks/common/IterTraits.h
@@ -55,7 +55,7 @@
 
 namespace StringPrivate {
 
-typedef unsigned int size_type;
+typedef std::size_t size_type;
 
 struct InpIterTag {};
 struct RndIterTag {};
diff --git a/sfworks/common/RefCntStorage.h b/sfworks/common/RefCntStorage.h
index 8a0f281..932ac5b 100644
--- a/sfworks/common/RefCntStorage.h
+++ b/sfworks/common/RefCntStorage.h
@@ -54,7 +54,7 @@
 
 namespace StringPrivate {
 
-typedef unsigned int size_type;
+typedef std::size_t size_type;
 
 template<typename E> struct RefCntData : public COMMON_NS::RefCounted<> {
     size_type   size_;
diff --git a/sfworks/common/SubAllocator.h b/sfworks/common/SubAllocator.h
index d89c085..9dd411b 100644
--- a/sfworks/common/SubAllocator.h
+++ b/sfworks/common/SubAllocator.h
@@ -55,7 +55,7 @@
 
 COMMON_NS_BEGIN
 
-typedef unsigned int size_type;
+typedef std::size_t size_type;
 
 COMMON_EXPIMP void suballoc_abort(const char* fmt, ...);
 
diff --git a/sfworks/common/impl/Debug.cxx b/sfworks/common/impl/Debug.cxx
index 6b7d4be..19617d0 100644
--- a/sfworks/common/impl/Debug.cxx
+++ b/sfworks/common/impl/Debug.cxx
@@ -49,7 +49,7 @@
 
 COMMON_NS_BEGIN
 
-typedef unsigned int size_type;
+typedef std::size_t size_type;
 using namespace std;
 
 struct TagCmp : public std::binary_function<const char*, const char*, bool> {
@@ -91,7 +91,7 @@ public:
         ThreadMutexClass::destroy(&os_lk_);
     }
 
-    typedef unsigned int size_type;
+    typedef std::size_t size_type;
 
     inline size_type getIndent() const { return indent_; }
     inline size_type setIndent(size_type newind)
-- 
1.7.5.1

